src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs

```diff
@@ -50,5 +53,17 @@ public ArgumentNullException(string? paramName, string? message)
         protected ArgumentNullException(SerializationInfo info, StreamingContext context) : base(info, context)
         {
         }
+
+        /// <summary>Throws an <see cref="ArgumentNullException"/> if <paramref name="argument"/> is null.</summary>
+        /// <param name="argument">The reference type argument to check for nullability.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        [NonVersionable]
```

Comment:
@jkotas, is this correct / worth doing, or should I remove it?

Reply:
I would remove it:

- The current compiler R2R is not able to create new metadata references that would be required to reference the other methods. It means `NonVersionable` on this method won't have any affect currently. 
- If this limitation is ever fixed, it would mean that `ThrowHelper.ThrowArgumentNullException` is de-facto public surface. If we were designing the public surface for this, I think it would be  `ArgumentNullException.Throw` and not `ThrowHelper.ThrowArgumentNullException`.